server/backend/check.ml

```diff
@@ -300,15 +319,18 @@ fi |} pkg pkg pkg (Server_configfile.platform_distribution conf)
       [String.concat "\n" [build; with_test ~conf pkg; with_lower_bound ~conf pkg; trailer]]
   | Server_configfile.Dune -> (
     let install_dune = "curl -fsSL https://get.dune.build/install | sh" in
-    [ install_dune ] @ prebuild_toolchains ~conf @ [
+    [ install_dune; install_remove_packages ] @ prebuild_toolchains ~conf @ [
       String.concat " && " [
         "cd $HOME";
         Printf.sprintf {|opam source %s|} pkg;
         Printf.sprintf {|cd %s|} pkg;
-        "opam install ./ --depext-only";
+        (* replace tarball opam metadata with more accurate opam repository metadata *)
+        "for opam in *.opam; do opam show --raw ${opam%.opam} > $opam; done";
+        remove_packages;
+        "opam install ./ --depext-only --with-test";
         set_up_workspace ~conf;
         Printf.sprintf {|%s dune pkg lock|} dune_path;
-        Printf.sprintf {|%s dune build --profile=release @install || echo "opam-health-check: Build failed" && exit 1|} dune_path]]
+        Printf.sprintf {|%s dune build || echo "opam-health-check: Build failed" && exit 1|} dune_path]]
```

Comment:
I still think `--profile=release` is preferable atm considering the number of failures caused by the warnings-are-errors default (unless you are looking for motivation to fix that)

Reply:
We could probably override it in the workspace but you're right that this is sensible. Added it again :)